### PR TITLE
Enable IdiomRecognition for warm methods

### PR DESCRIPTION
**Pull request by yanluo7**
Signed-off-by: Yan Luo <yan_luo@ca.ibm.com>

### CODE REVIEW

File: runtime/tr.source/trj9/optimizer/J9Optimizer.cpp [hunk 1]
```diff
@@ -665,6 +665,7 @@ static const OptimizationStrategy cheapWarmStrategyOpts[] =
    { OMR::blockShuffling                                                             },
    { OMR::idiomRecognition,                          OMR::IfLoopsAndNotProfiling     },
```
Comment (DanHeidinga) [inline runtime/tr.source/trj9/optimizer/J9Optimizer.cpp]: Is this `idiomRecognition` intended to be duplicated on s390 or should it have been deleted from the #ifdef block?

Comment (andrewcraik) [reply runtime/tr.source/trj9/optimizer/J9Optimizer.cpp]: This shouldn't be duplicated it should be pulled out of the ifdef

**Comment by andrewcraik** [comment]
We will need to understand the performance implications of this change before merging

File: runtime/tr.source/trj9/optimizer/J9Optimizer.cpp (hunk 1 above)
Comment (fjeremic) [reply runtime/tr.source/trj9/optimizer/J9Optimizer.cpp]: Good catch @DanHeidinga. It should **not** be duplicated.

**Comment by andrewcraik** [comment]
@yanluo7 can you please address the review comments and comment on the performance testing done?

**Comment by yanluo7** [comment]
Pulled the code out of Z ifdef so it's enabled on all platforms.
We've measured performance metrics for several benchmarks:

- For liberty, no impact on startup time or throughput.
- For WAS, no impact on startup time, 0.5% improvement on throughput.
- For Open Liberty ACMEAIR benchmark, no impact on compilation time, ~1% improvement on throughput.

**Comment by andrewcraik** [comment]
Jenkins test sanity

**Comment by andrewcraik** [comment]
Jenkins test sanity

**Pull request merged**